_source/_posts/2023-07-28-scim-workshop.md

```diff
@@ -0,0 +1,1144 @@
+
+### View requests to and from the app
+When [ngrok](https://ngrok.com/) is supported in my environment, it's my favorite tool for viewing requests to and from a local server. If ngrok is blocked, use Okta's [system logs](https://help.okta.com/en-us/Content/Topics/Reports/Reports_SysLog.htm) to see requests from the app to Okta. [Morgan](https://github.com/expressjs/morgan) can help you log inbound HTTP requests to the server.
+
+[Wireshark](https://www.wireshark.org/) is a powerful tool for examining all network traffic, including HTTP requests. 
+
+### Public URLs for local apps
+ngrok is my favorite solution for giving an app a public address, but it's not allowed by all networks. When ngrok isn't supported, [localtunnel]() is a good alternative. 
+### Testing Server Behavior
+[Postman](https://www.postman.com/) and [Hoppscotch](https://hoppscotch.io/) offer a friendly interface for sending custom request bodies and headers to a server. If you prefer to work in the terminal, cURL can do the same things with the right arguments passed to it. 
+### Debugging Running Code
+If you are using Visual Studio Code as your IDE, I recommend enabling your built-in debugger tool. This will allow you to inspect the request body by adding breakpoints as needed. 
+
+Follow [Visual Studio's docs](https://code.visualstudio.com/docs/editor/debugging) to find the Run And Debug view. I used the following `launch.json` configuration for Visual Studio when working on this app:
+
+```
+{
+    "configurations": [
+        {
+            "name": "Debug Node App",
+            "type": "node",
+            "request": "attach",
+            "port": 3333
+        },
+    ]
+}
+```
+Replace the Todo app's `apps/api/webpack.config.js` with the following to facilitate debugging through Visual Studio: 
+
+```
+const { composePlugins, withNx } = require('@nx/webpack');
+const path = require('path')
+
+// Nx plugins for webpack.
+module.exports = composePlugins(withNx(), (config) => {
+    // Update the webpack config as needed here.
+    // e.g. `config.plugins.push(new MyPlugin())`
+    config.output.devtoolModuleFilenameTemplate = function (info) {
+        const rel = path.relative(process.cwd(), info.absoluteResourcePath);
+        return `webpack:///./${rel}`
+    }
+    return config
+});
+```
+
+## Conclusion 
+
+Congratulations, you've added SCIM support to an OIDC application! Not only have you supported SCIM for a single Okta organization, but the implementation choices made throughout this workshop prioritize scalability so that the Todo app can integrate with as many additional identity providers as necessary! By giving every integration its own unique SCIM endpoint and API token, you prepare your application to seamlessly integrate with as many customers as you want. 
```

Comment:
It's worth pointing out here that some identity providers that support SCIM may have implemented a slightly different subset of SCIM than Okta implements, so you will need to read the documentation for the specific provider when adding support for a second IDP.